src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -188,6 +191,77 @@ func (c controller) PostInstallConfigs(wg *sync.WaitGroup) {
 	c.sendCompleteInstallation(true, "")
 }
 
+func (c controller) UpdateBMHs(wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		time.Sleep(GeneralWaitTimeout)
+		exists, err := c.kc.IsMetalProvisioningExists()
+		if err == nil && exists {
+			c.log.Infof("Provisioning CR exists, no need to update BMHs")
+			return
+		}
+
+		bmhs, err := c.kc.ListBMHs()
+		if err != nil {
+			c.log.WithError(err).Errorf("Failed to BMH hosts")
+			continue
+		}
+
+		allUpdated := c.updateBMHStatus(bmhs)
+		if allUpdated {
+			c.log.Infof("Updated all the BMH CRs, finished successfully")
+			return
+		}
+	}
+}
+
+func (c controller) updateBMHStatus(bmhList metal3v1alpha1.BareMetalHostList) bool {
```

Comment:
No one is updating this status? in parallel ?

Reply:
no. This code is called in case provisioning for metal3 is not defined, in whch case BMO is not running, and it is the code that updates the statuses